source/parser/function.civet

```diff
@@ -575,25 +575,36 @@ function expressionizeIteration(exp: IterationExpression): void
     updateParentPointers exp
     return
 
-  exp.resultsRef ??= makeRef "results"
-  { resultsRef } := exp
+  if star
+    assignResults block, (node) =>
+      [ "yield ", node ]
+    braceBlock(block)
+    children.splice(i,
+      1,
+      wrapIIFE(children.map((c) => ["", c, undefined]), async, star)
+    )
+    updateParentPointers block // <- is this necessary? I do add `"yield "`
```

Comment:
Yes, I think so. It could maybe be moved to `assignResults`. But for now probably makes sense to match the old code. (This could be brought outside the `if`.)

Reply:
It’s not done on the same AST node however